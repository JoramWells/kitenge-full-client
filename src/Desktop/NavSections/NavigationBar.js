import React, { useState } from "react";
// import LeftMenu from "./NavSections/LeftMenu";
import RightMenu from "./RightMenu";
import { message } from "antd";
import { Link, withRouter } from "react-router-dom";
import { searchItems } from "../../_actions/searchActions";
import { useDispatch } from "react-redux";
import {
  SearchIcon,
  XIcon,
  MenuIcon,
  CreditCardIcon,
  ClockIcon,
  ThumbUpIcon,
  DotsCircleHorizontalIcon,
  HomeIcon,
} from "@heroicons/react/solid";
import styled from "styled-components";

function NavigationBar({ props, activateOption }) {
  const [visible, setVisible] = useState("hidden");
  const [sidebar, setSidebar] = useState(false);
  const activate = () => setSidebar(!sidebar);
  const dispatch = useDispatch();
  async function onSearch() {
    dispatch(searchItems({ keyword: "amazing", min_videos: "1" }));
    message.info("Waaaaaaat!!!");
    // await axios.post('/andeyo',{keyword:"amazing",min_videos:1}).then(response=>{
    //   setPosts(response.data)
    //   console.log(posts)
    // }).catch(err=>console.log(err))
    setTimeout(() => {
      props.history.push("/searched");
    }, 1000);
  }

  return (
    <>
      <Navbar className="desktop__navbar ">
        <div className="p-2 flex flex-row space-x-2 items-center">
          <MenuIcon
            className="h-5 text-gray-800"
            onClick={activateOption ? activate : null}
          />
          <Link to="/" className="text-gray-500 font-semibold text-xl">
            Do3ens
          </Link>
          {/* <Row justify="center">
          <Col span={6}>
            <Search placeholder="Search.." onSearch={onSearch} />
          </Col>
        </Row> */}
        </div>

        <div
          style={{ width: "30rem" }}
          className="bg-gray-100 flex flex-row items-center content-center space-x-2 rounded-full"
        >
          <SearchIcon className="h-10 text-gray-400 p-1.5" />
          <input
            placeholder="Search..."
            className=" text-sm w-full focus:outline-none bg-transparent"
            onClick={() => setVisible("visible")}
          />
          <XIcon className="h-10 text-gray-400 p-1.5" />
        </div>
        <div
          style={{
            position: "fixed",
            width: "50%",
            height: "200px",
            top: "53px",
            backgroundColor: "white",
            zIndex: "4",
            left: "50%",
            right: "50%",
            transform: " translate(-50%, 0%)",
            visibility: visible,
          }}
          className="rounded-md shadow-md p-2"
        >
          hdsjhdsjd
          <XIcon className="h-5" onClick={()=>setVisible("hidden")} />
        </div>

        <div>
          <RightMenu mode="horizontal" />
        </div>
      </Navbar>
      <Sidenav
        className={
          sidebar
            ? "nav-menu active flex flex-row items-center content-center"
            : "nav-menu"
        }
      >
        <ul className="leading-8 ">
          <Li>
            <HomeIcon className="h-5 ml-3 -mt-4 text-gray-400 mr-4" />
            <p className=" text-sm text-gray-600">Home</p>
          </Li>

          <hr className="text-white p-2" />
          <Li>
            <CreditCardIcon className="h-5 ml-3 -mt-4 text-gray-400 mr-4" />
            <p className="text-sm text-gray-600 ">Purchases </p>
          </Li>

          <Li>
            <ClockIcon className="h-5 ml-3 -mt-3 text-gray-400 mr-4" />
            <p className="text-sm text-gray-500 ">Recent purchases </p>
          </Li>
          <hr className="text-white py-2" />
          <Li>
            <ThumbUpIcon className="h-5 ml-3 -mt-4 text-gray-400 mr-4" />
            <p className="text-sm text-gray-600 ">Liked Items</p>
          </Li>
          <Li>
            <DotsCircleHorizontalIcon className="h-5 ml-3 -mt-3 text-gray-400 mr-4" />
            <p className="text-sm text-gray-600 ">More Items</p>
          </Li>
        </ul>
      </Sidenav>
    </>
  );
}

export default withRouter(NavigationBar);

const Navbar = styled.nav`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  position: fixed;
  width: 100%;
  top: 0%;
  z-index: 20;
  padding: 0.3rem;
  background-color: white;
`;

const Sidenav = styled.nav`
  padding-top: 3.77rem;
  width: 17%;
  height: 100%;
  background-color: whitesmoke;
  display: fixed;
  z-index: 1;
  text-align: center;
`;

const Li = styled.li`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: start;
  align-content: center;
  padding: 0.2rem;
  :hover {
    cursor: pointer;
    color: gray;
    /* background-color: white; */
  }
  color: black;
  margin: 0.3rem;
`;
