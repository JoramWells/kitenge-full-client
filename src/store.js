import thunk from "redux-thunk";
import Cookie from "js-cookie";

import {
//   categoryListReducer,
//   categoryListsReducer,
//   productDeleteReducer,
//   productDetailReducer,
  productListReducer,
//   productSaveReducer,
//   productUpdateReducer,
} from "./_reducers/productListReducer";
import { applyMiddleware, combineReducers, compose, createStore } from "redux";
// import {
//   userGoogleReducer,
//   userRegisterReducer,
//   userSigninReducer,
// } from "./_reducers/userReducers";
// import { cartReducer } from "./_reducers/cartReducers";
// import {
//   confirmPaymentReducer,
//   paymentListReducer,
//   paymentReducer,
// } from "./_reducers/paymentReducer";

const cartItems = Cookie.getJSON("cartItems") || [];
const userInfo = Cookie.getJSON("userInfo") || null;
const initialState = { cart: { cartItems }, userSignin: { userInfo } };

const reducer = combineReducers({
  productList: productListReducer,
//   productDetail: productDetailReducer,
//   userSignin: userSigninReducer,
//   userRegister: userRegisterReducer,
//   productSave: productSaveReducer,
//   cart: cartReducer,
//   categoryList: categoryListReducer,
//   categoryLists: categoryListsReducer,
//   productUpdate: productUpdateReducer,
//   productDelete: productDeleteReducer,
//   googleUser: userGoogleReducer,
//   payment: paymentReducer,
//   confirmDetails: confirmPaymentReducer,
//   paymentList: paymentListReducer,
});
const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(
  reducer,
  initialState,
  composeEnhancer(applyMiddleware(thunk))
);

export default store;
