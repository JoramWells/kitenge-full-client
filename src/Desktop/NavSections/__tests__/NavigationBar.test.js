import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import { render, cleanup } from "@testing-library/react";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import NavigationBar from "../NavigationBar";
import ReactDOM from "react-dom";
import store from "../../../store";

// const setup = (props = {}) => {
//   const component = shallow(<NavigationBar {...props} />);
//   return component;
// };
afterEach(cleanup);
describe("Navigation bar", () => {
  //   let component;
  //   beforeEach(() => {
  //     component = setup();
  //   });
  //   it("Should render without failing", () => {
  //     const wrapper = component.find(".desktop__navbar");
  //     expect(wrapper).toBeInTheDocument();
  //   });

  it("renders Navbar correctly", () => {
    const nav = document.createElement("div");
    ReactDOM.render(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <NavigationBar />

            <Switch></Switch>
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
      nav
    );
  });
});
