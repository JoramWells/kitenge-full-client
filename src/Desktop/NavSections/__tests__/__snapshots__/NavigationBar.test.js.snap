// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar matches snapshot 1`] = `
Array [
  <nav
    className="sc-bdnxRM bBqHQE desktop__navbar shadow-lg"
    data-testid="nav"
  >
    <div
      className="p-2 flex flex-row space-x-2 items-center"
    >
      <svg
        className="h-5 text-gray-800"
        fill="currentColor"
        onClick={null}
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
          fillRule="evenodd"
        />
      </svg>
      <a
        className="text-gray-700 font-semibold text-xl"
        href="/"
        onClick={[Function]}
      >
        Do3ensKE
      </a>
    </div>
    <div
      className="bg-gray-100 flex flex-row items-center content-center space-x-2 rounded-full"
      style={
        Object {
          "width": "30rem",
        }
      }
    >
      <svg
        className="h-10 text-gray-400 p-1.5"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
          fillRule="evenodd"
        />
      </svg>
      <input
        className=" text-sm w-full focus:outline-none bg-transparent"
        onClick={[Function]}
        placeholder="Search..."
      />
      <svg
        className="h-10 text-gray-400 p-1.5"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div>
      <div
        className="hover:cursor-pointer py-2 px-4 border  focus:outline-none rounded-md text-gray-900"
        onClick={[Function]}
      >
        Login
      </div>
    </div>
  </nav>,
  <nav
    className="sc-gtsrHT iLPqLW nav-menu"
  >
    <ul
      className="leading-8 "
    >
      <li
        className="sc-dlnjwi hHwtZA"
      >
        <svg
          className="h-5 ml-3 text-gray-500 mr-4"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"
          />
        </svg>
        <div
          className=" text-sm text-gray-700"
        >
          Home
        </div>
      </li>
      <hr
        className="text-white p-2"
      />
      <li
        className="sc-dlnjwi hHwtZA"
      >
        <svg
          className="h-5 ml-3  text-gray-500 mr-4"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 4a2 2 0 00-2 2v1h16V6a2 2 0 00-2-2H4z"
          />
          <path
            clipRule="evenodd"
            d="M18 9H2v5a2 2 0 002 2h12a2 2 0 002-2V9zM4 13a1 1 0 011-1h1a1 1 0 110 2H5a1 1 0 01-1-1zm5-1a1 1 0 100 2h1a1 1 0 100-2H9z"
            fillRule="evenodd"
          />
        </svg>
        <div
          className="text-sm text-gray-700"
        >
          Purchases 
        </div>
      </li>
      <li
        className="sc-dlnjwi hHwtZA"
      >
        <svg
          className="h-5 ml-3  text-gray-500 mr-4"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10 2a4 4 0 00-4 4v1H5a1 1 0 00-.994.89l-1 9A1 1 0 004 18h12a1 1 0 00.994-1.11l-1-9A1 1 0 0015 7h-1V6a4 4 0 00-4-4zm2 5V6a2 2 0 10-4 0v1h4zm-6 3a1 1 0 112 0 1 1 0 01-2 0zm7-1a1 1 0 100 2 1 1 0 000-2z"
            fillRule="evenodd"
          />
        </svg>
        <div
          className="text-sm text-gray-700"
        >
          Your products 
        </div>
      </li>
      <li
        className="sc-dlnjwi hHwtZA"
      >
        <svg
          className="h-5 ml-3 text-gray-500 mr-4"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10 18a8 8 0 100-16 8 8 0 000 16zm1-12a1 1 0 10-2 0v4a1 1 0 00.293.707l2.828 2.829a1 1 0 101.415-1.415L11 9.586V6z"
            fillRule="evenodd"
          />
        </svg>
        <div
          className="text-sm text-gray-700"
        >
          Recent purchases 
        </div>
      </li>
      <hr
        className="text-white py-2"
      />
      <li
        className="sc-dlnjwi hHwtZA"
      >
        <svg
          className="h-5 ml-3  text-gray-500 mr-4"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 10.5a1.5 1.5 0 113 0v6a1.5 1.5 0 01-3 0v-6zM6 10.333v5.43a2 2 0 001.106 1.79l.05.025A4 4 0 008.943 18h5.416a2 2 0 001.962-1.608l1.2-6A2 2 0 0015.56 8H12V4a2 2 0 00-2-2 1 1 0 00-1 1v.667a4 4 0 01-.8 2.4L6.8 7.933a4 4 0 00-.8 2.4z"
          />
        </svg>
        <div
          className="text-sm  text-gray-700"
        >
          Liked Items
        </div>
      </li>
      <li
        className="sc-dlnjwi hHwtZA"
      >
        <svg
          className="h-5 ml-3 text-gray-500 mr-4"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M10 18a8 8 0 100-16 8 8 0 000 16zM7 9H5v2h2V9zm8 0h-2v2h2V9zM9 9h2v2H9V9z"
            fillRule="evenodd"
          />
        </svg>
        <div
          className="text-sm  text-gray-700"
        >
          More Items
        </div>
      </li>
    </ul>
  </nav>,
]
`;
